### ISSUE TITLE
Questions about adversarial textures

### ISSUE THREAD

**Issue by leige521** [Gandolfczjh/3D2Fool#20]
Hello Dear Author, I would like to ask if the texture seed image file, such as 'texture_seed.png', you provided was obtained by randomly generating adversarial patches and iteratively optimising them with the following code?
```
# continuous color
camou_para = torch.rand([1, h, w, 3]).float().to(args.device)
camou_para.requires_grad_(True)
optimizer = optim.Adam([camou_para], lr=args.lr)
camou_para1 = expand_kernel(camou_para.permute(0, 3, 1, 2)).permute(0, 2, 3, 1)
```
Or did you use other images as templates to iteratively optimise them?

**Comment by Gandolfczjh** [comment]
Yes. We use this code to initialize the texture seed and then proceed with optimization training. We also experimented with other initialization methods but found that they had little impact on the results, so only random initialization is presented here.

**Comment by leige521** [comment]
I see, thank you for your wonderful answer, and i will close the comment
